core/json-utils/src/main/java/software/amazon/awssdk/protocols/jsoncore/JsonWriter.java

```diff
@@ -179,13 +180,31 @@ private Builder() {
          * The {@link JsonFactory} implementation to be used when parsing the input. This allows JSON extensions like CBOR or
          * Ion to be supported.
          *
-         * <p>It's highly recommended us use a shared {@code JsonFactory} where possible, so they should be stored statically:
+         * <p>It's highly recommended to use a shared {@code JsonFactory} where possible, so they should be stored statically:
          * http://wiki.fasterxml.com/JacksonBestPracticesPerformance
          *
          * <p>By default, this is {@link JsonNodeParser#DEFAULT_JSON_FACTORY}.
+         *
+         * <p>Setting this value will also override any values set via {@link #jsonGeneratorFactory}.
          */
         public JsonWriter.Builder jsonFactory(JsonFactory jsonFactory) {
-            this.jsonFactory = jsonFactory;
+            jsonGeneratorFactory(jsonFactory::createGenerator);
+            return this;
+        }
+
+        /**
+         * A factory for {@link JsonGenerator}s based on an {@link OutputStream}. This allows custom JSON generator
+         * configuration, like pretty-printing output.
+         *
+         * <p>It's highly recommended to use a shared {@code JsonFactory} within this generator factory, where possible, so they
+         * should be stored statically: http://wiki.fasterxml.com/JacksonBestPracticesPerformance
+         *
+         * <p>By default, this delegates to {@link JsonNodeParser#DEFAULT_JSON_FACTORY} to create the generator.
+         *
+         * <p>Setting this value will also override any values set via {@link #jsonFactory}.
+         */
+        public JsonWriter.Builder jsonGeneratorFactory(JsonGeneratorFactory jsonGeneratorFactory) {
```

Comment:
Added this to allow me to configure the underlying JSON generator to pretty-print the JSON.